zig-src/beam.zig

```diff
@@ -156,45 +156,26 @@ const raw_beam_allocator_vtable = Allocator.VTable{
     .free = raw_beam_free,
 };
 
-fn raw_beam_alloc(
-    _: *anyopaque,
-    len: usize,
-    ptr_align: u29,
-    _: u29,
-    _: usize,
-) Allocator.Error![]u8 {
+fn raw_beam_alloc(_: *anyopaque, len: usize, ptr_align: u8, _: usize) ?[*]u8 {
     if (ptr_align > MAX_ALIGN) {
-        return error.OutOfMemory;
+        return null;
     }
-    const ptr = e.enif_alloc(len) orelse return error.OutOfMemory;
-    return @ptrCast([*]u8, ptr)[0..len];
+    const ptr = e.enif_alloc(len) orelse return null;
+    return @as([*]u8, @ptrCast(ptr));
 }
 
-fn raw_beam_resize(
-    _: *anyopaque,
-    buf: []u8,
-    _: u29,
-    new_len: usize,
-    _: u29,
-    _: usize,
-) ?usize {
+fn raw_beam_resize(_: *anyopaque, buf: []u8, _: u8, new_len: usize, _: usize) bool {
     if (new_len == 0) {
         e.enif_free(buf.ptr);
-        return 0;
+        return true;
     }
     if (new_len <= buf.len) {
-        return new_len;
+        return true;
     }
-    // Is this the right thing to do???
-    return null;
+    return false;
 }
```

Comment:
@ityonemo could you have a look at this change? It was originally from Zigler and being updated to Zig 0.11